app/src/main/kotlin/com/simplemobiletools/calculator/helpers/GetCryptoTask.kt

```diff
@@ -0,0 +1,62 @@
+package com.simplemobiletools.calculator.helpers
+
+import android.os.AsyncTask
+import android.util.Log
+import com.simpletools.calculator.commons.helpers.Calculator
+import com.simpletools.calculator.commons.helpers.Formatter
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.BufferedInputStream
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+class GetCryptoTask(val cryptoFROM: String, val cryptoTO: String, cryptoActivity: Calculator, cryptoCalculatorImpl: CryptoCalculatorImpl) : AsyncTask<Void, Void, String>() {
+
+    var mCalc: Calculator = cryptoActivity
+    var cryptoCalculator: CryptoCalculatorImpl = cryptoCalculatorImpl
+
+    override fun doInBackground(vararg params: Void?): String? {
+        // old API that crashes after too many requests, keeping
+        //val url = URL("https://api.cryptonator.com/api/ticker/$cryptoFROM-$cryptoTO")
+        val url = URL("https://min-api.cryptocompare.com/data/price?fsym=$cryptoFROM&tsyms=$cryptoTO")
+
+        val httpClient = url.openConnection() as HttpURLConnection
+
+        if (httpClient.responseCode == HttpURLConnection.HTTP_OK) {
```

Comment:
Error handling should go here add a switch case for HttpURLConnection.HTTP_NOT_FOUND and trigger an alert dialog in the ui thread.

Reply:
Will add more error handling for this.